scala3doc/src/main/scala/dotty/dokka/tasty/TastyParser.scala

```diff
@@ -0,0 +1,205 @@
+package dotty.dokka
+package tasty
+
+import org.jetbrains.dokka.plugability._
+import org.jetbrains.dokka.transformers.sources._
+
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.model._
+import org.jetbrains.dokka.links._
+import org.jetbrains.dokka.model.doc._
+import org.jetbrains.dokka.base.parsers._
+import org.jetbrains.dokka.plugability.DokkaContext
+import collection.JavaConverters._
+import org.jetbrains.dokka.model.properties.PropertyContainer
+import org.jetbrains.dokka.model.properties.PropertyContainerKt._
+import org.jetbrains.dokka.model.properties.{WithExtraProperties}
+import java.util.{List => JList}
+
+import scala.tasty.Reflection
+import scala.tasty.inspector.TastyInspector
+import dotty.dokka.model.api.withNewMembers
+import com.virtuslab.dokka.site.SourceSetWrapper
+
+/** Responsible for collectively inspecting all the Tasty files we're interested in.
+  *
+  * Delegates most of the work to [[TastyParser]] [[dotty.dokka.tasty.TastyParser]].
+  */
+case class DokkaTastyInspector(sourceSet: SourceSetWrapper, parser: Parser, config: DottyDokkaConfig) extends DokkaBaseTastyInspector with TastyInspector
+
+import dotty.tools.dotc.core.Contexts.{Context => DottyContext}
+case class SbtDokkaTastyInspector(
+  sourceSet: SourceSetWrapper,
+  config: DottyDokkaConfig,
+  filesToDocument: List[String],
+  rootCtx: DottyContext,
+) extends DokkaBaseTastyInspector:
+  self =>
+
+  import dotty.tools.dotc.Compiler
+  import dotty.tools.dotc.Driver
+  import dotty.tools.dotc.Run
+  import dotty.tools.dotc.core.Contexts.Context
+  import dotty.tools.dotc.core.Mode
+  import dotty.tools.dotc.core.Phases.Phase
+  import dotty.tools.dotc.fromtasty._
+  import dotty.tools.dotc.quoted.QuoteContextImpl
+
+  val parser: Parser = null
+
+  def run(): List[DPackage] = {
+    val driver = new InspectorDriver
+    driver.run(filesToDocument)(rootCtx)
+    result()
+  }
+
+  class InspectorDriver extends Driver:
+    override protected def newCompiler(implicit ctx: Context): Compiler = new TastyFromClass
+
+    def run(filesToDocument: List[String])(implicit ctx: Context): Unit =
+      doCompile(newCompiler, filesToDocument)
```

Comment:
What can `filesToDocument` contain? `.tasty` and `.jar` file? If so it is ok.

Reply:
List of `.tasty` files. We do support jars however we extract them to temporary directory and use extracted `.tasy` files. I think we will change that once #10061 is ready